### ISSUE TITLE
issue on mpt3sas kernel module (unable to file bug upstream)

### ISSUE THREAD

**Issue by zeph** [linuxmint/cinnamon#12251]
### Distribution

Cinnamon 6.0.4

### Package version

kernel 6.1.0-21-amd64

### Frequency

Always

### Bug description

Subject: mpt3sas.max_queue_depth=10000 being ignored by mpt3sas module
Date: Jun 21 2024, at 5:38 pm
To: [DL-MPTFusionLinux@lsi.com](mailto:DL-MPTFusionLinux@lsi.com), [MPT-FusionLinux.pdl@avagotech.com](mailto:MPT-FusionLinux.pdl@avagotech.com)
Hi, I got the pleasure of stumbling on the Queue size issue everyone
 seems to be having with the card LSI 9210-8i
https://bbs.archlinux.org/viewtopic.php?pid=1967144#p1967144

There are tons of internet forum posting that to have the card able to list the devices I have to change the max_queue_depth , but the module seems to be blatantly ignoring such change. I added it in modprobe.d (uneffected, yes I did update initramf) and at the GRUB kernel boot param level.

I even got to run this fancy script... leading to the following output

$ cat /proc/modules | cut -f 1 -d " " | while read module; do  echo "Module: $module";  if [ -d "/sys/module/$module/parameters" ]; then   ls /sys/module/$module/parameters/ | while read parameter; do    echo -n "Parameter: $parameter --> ";    cat /sys/module/$module/parameters/$parameter;   done;  fi;  echo; done

Module: mpt3sas
Parameter: diag_buffer_enable --> -1
Parameter: disable_discovery --> -1
Parameter: enable_sdev_max_qd --> N
Parameter: hbas_to_enumerate --> 0
Parameter: host_tagset_enable --> 1
Parameter: irqpoll_weight --> -1
Parameter: logging_level --> 0
Parameter: max_lun --> 16895
Parameter: max_msix_vectors --> -1
Parameter: max_queue_depth --> 10000
Parameter: max_sectors --> 65535
Parameter: max_sgl_entries --> -1
Parameter: missing_delay --> -1,-1
Parameter: mpt3sas_fwfault_debug --> 0
Parameter: msix_disable --> -1
Parameter: perf_mode --> -1
Parameter: poll_queues --> 0
Parameter: prot_mask --> -1
Parameter: smp_affinity_enable --> 1
 
The value is set, but the value is still low...

$ dmesg |grep Queue
[    1.388586] mpt2sas_cm0: Current Controller Queue Depth(3364),Max Controller Queue Depth(3432)

I'm currently running on a Linux MINT LXDE with kernel 6.1.0-21-amd64

Any idea how to proceed? Shall I attempt a different param or it won't effect the behaviour?

regards,

### Steps to reproduce

just see email above

### Expected behavior

expected lsscsi to list the tape drive

### Additional information

I tried to file the bug upstream to the 2 email mentioned at kernel level, but they bounced

**Comment by leigh123linux** [comment]
Not our problem.

**Comment by zeph** [comment]
@leigh123linux I appreciate your little support, but the Linux Kernel mailing list seems to be aiming directly to the distros to file for bugs, so... at least you have a better recommendation how to file such? Debian perhaps?

**Comment by zeph** [comment]
btw, before closing, check what they state on their tracking system https://bugzilla.kernel.org/

**Comment by leigh123linux** [comment]
> @leigh123linux I appreciate your little support, but the Linux Kernel mailing list seems to be aiming directly to the distros to file for bugs, so... at least you have a better recommendation how to file such? Debian perhaps?


Fair enough, but linuxmint doesn't do any kernel development, they use the ubuntu or debian kernel.
Report the issue to ubuntu, debian
